examples/distro/agent/build.gradle

```diff
@@ -1,46 +1,121 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+
 plugins {
-  id("com.github.johnrengelman.shadow") version "7.1.2"
+  id("com.github.johnrengelman.shadow")
 }
 
 apply from: "$rootDir/gradle/shadow.gradle"
 
 def relocatePackages = ext.relocatePackages
 
 configurations {
-  customShadow
+  // this configuration collects libs that will be placed in the bootstrap classloader
+  bootstrapLibs {
+    canBeResolved = true
+    canBeConsumed = false
+  }
+  // this configuration collects libs that will be placed in the agent classloader, isolated from the instrumented application code
+  javaagentLibs {
+    canBeResolved = true
+    canBeConsumed = false
+  }
+  // this configuration stores the upstream agent dep that's extended by this project
+  upstreamAgent {
+    canBeResolved = true
+    canBeConsumed = false
+  }
 }
 
 dependencies {
-  customShadow project(path: ":custom", configuration: "shadow")
-  customShadow project(path: ":instrumentation", configuration: "shadow")
-  implementation "io.opentelemetry.javaagent:opentelemetry-javaagent:${versions.opentelemetryJavaagent}"
+  bootstrapLibs(project(":bootstrap"))
+
+  javaagentLibs(project(":custom"))
+
+  upstreamAgent("io.opentelemetry.javaagent:opentelemetry-javaagent:${versions.opentelemetryJavaagent}")
 }
 
-CopySpec isolateSpec() {
+// collect all instrumentation sub projects
+Project instr_project = project
+DependencyHandler javaagentDependencies = dependencies
+project(":instrumentation").subprojects {
+  afterEvaluate { Project subProj ->
+    if (subProj.getPlugins().hasPlugin('java')) {
+      javaagentDependencies.add(instr_project.configurations.javaagentLibs.name, project(subProj.path))
+    }
+  }
+}
```

Comment:
Nit: since it's supposed to be a demo/showcase of a javaagent distro, how about replacing this with a simple `javaagentLibs(project(":instrumentation:servlet-3"))` in the `dependencies` section? I'm sure more advanced gradle users can figure out how to auto-find instrumentation modules themselves; we should just be showing the general javaagent setup.

Reply:
good point